From d37bc00ef3646af2bbff46a86bd868f6712c448a Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 25 Jul 2012 22:52:17 +0800
Subject: [PATCH 315/498] ENGR00217633 camera add force contiguous usage flag

camera add force contiguous usage flag to make sure allocate continue memory

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 mx6/libcamera/CameraHal.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index 2e5400f..dd7d565 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -784,9 +784,9 @@ namespace android {
             return NO_ERROR;//BAD_VALUE;
         }
 
-        //Make sure the buffer be phiscal continuous
+        //Make sure the buffer be physical continuous
         err = mNativeWindow->set_usage(mNativeWindow,
-                GRALLOC_USAGE_SW_READ_NEVER | GRALLOC_USAGE_HW_TEXTURE);
+                GRALLOC_USAGE_SW_READ_NEVER | GRALLOC_USAGE_FORCE_CONTIGUOUS | GRALLOC_USAGE_HW_TEXTURE);
         if(err != 0){
             CAMERA_LOG_ERR("native_window_set_usage failed:%s(%d)",
                     strerror(-err), -err);
-- 
1.8.0

